fastplotlib/layouts/_gridplot.py

```diff
@@ -92,12 +92,35 @@ def __init__(
                 self.shape
             )
 
+        if "names" in kwargs.keys():
+            self.names = to_array(kwargs["names"])
+            if self.names.shape != self.shape:
+                raise ValueError(f"subplot names: {self.names} must be in gridplot shape: {self.shape}")
+        else:
+            self.names = None
```

Comment:
we should probably just add `names` as a kwarg with default value of `None`. Allow it to be either list of list of `str` or array of `str`

Reply:
And then add a sanity check to make sure if `names` is provided that it conforms to the types

Reply:
Perfect